File: Modules/hospital/Speciality/SpecialityColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace Serene2.hospital.Columns;

[ColumnsScript("hospital.Speciality")]
[BasedOnRow(typeof(SpecialityRow), CheckNames = true)]
public class SpecialityColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignCenter]
    public int SpecialityId { get; set; }
    [EditLink]
    public string SpecialityName { get; set; }

    public string DoctorName { get; set; }
}

File: Modules/ServerTypes/hospital/SpecialityColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SpecialityRow } from "./SpecialityRow";

export interface SpecialityColumns {
    SpecialityId: Column<SpecialityRow>;
    SpecialityName: Column<SpecialityRow>;
    DoctorName: Column<SpecialityRow>;
}

export class SpecialityColumns extends ColumnsBase<SpecialityRow> {
    static readonly columnsKey = 'hospital.Speciality';
    static readonly Fields = fieldsProxy<SpecialityColumns>();
}